### PR TITLE
Clean up display of bare URLs

### PR DESCRIPTION
**Pull request by gfontenot**
- Truncate to 30 characters
 - Remove superfluous leading characters
 - Don't add linebreaks inside a URL

See individual commits for more info.

### CODE REVIEW

File: static/js/app.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+function truncateURLs() {
+  $('.scream a').each(function() {
+    displayTitle = this.innerText
+    if (displayTitle.startsWith('http')) {
+      this.innerText = displayTitle
+        .replace(/https?:\/\/(w{3}.)?/g, '')
+        .truncate(30)
+
+      this.className += ' nowrap'
```
Comment (BlakeWilliams) [inline static/js/app.js]: jQuery? `$(this).addClass('nowrap')`?

**Comment by edwardloveall** [comment]
It's worth noting that this can mostly be done with css:

```css
a {
  display: inline-block;
  max-width: 18ex;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}
```

http://codepen.io/edwardloveall/pen/aJBYRP

It does not, however, remove the `http(s)`

**Comment by gfontenot** [comment]
The reason I did _not_ go with the pure css solution:

 - I only want to apply this effect to bare urls, not _every_ link
 - I wanted to perform more cleanup (removing `https?://(w{3}.)?`)
 - ok tbh i could not figure out how to get it to work

File: static/js/app.js (hunk 1 above)
Comment (gfontenot) [reply static/js/app.js]: yeah, I had that, went back and forth on it.

**Pull request merged**